addons/binding/org.openhab.binding.jeelink/src/main/java/org/openhab/binding/jeelink/handler/JeeLinkReadingListener.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.jeelink.handler;
+
+/**
+ * Base class for a reading listener.
+ *
+ * @author Volker Bier - Initial contribution
+ */
+public class JeeLinkReadingListener<R extends Reading> {
```

Comment:
It is not abstract, as it provides functionality to keep track of the current reading and can be used as is. I could try to improve the apidoc to make that clear.